From 6fab8fabcac27ed65ad3fba3e7631bbcbef6b6a4 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 27 Oct 2021 15:16:27 +1000
Subject: [PATCH] meson: add a libv4l2-path option to match libjack-path

pw-v4l2 uses the gnu ld's ${LIB} features, see meson.build.
Make it possible to work around this by specifying an explicit path,
just like for pw-jac.

Fixes #1751
---
 meson_options.txt             | 3 +++
 pipewire-v4l2/src/meson.build | 9 +++++++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/meson_options.txt b/meson_options.txt
index f10a92fd4..06d964316 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -61,6 +61,9 @@ option('jack-devel',
 option('libjack-path',
        description: 'Where to install the libjack.so library',
        type: 'string')
+option('libv4l2-path',
+       description: 'Where to install the libpw-v4l2.so library',
+       type: 'string')
 option('spa-plugins',
        description: 'Enable spa plugins integration',
        type: 'feature',
diff --git a/pipewire-v4l2/src/meson.build b/pipewire-v4l2/src/meson.build
index 9675aea4b..a3c97c189 100644
--- a/pipewire-v4l2/src/meson.build
+++ b/pipewire-v4l2/src/meson.build
@@ -15,8 +15,13 @@ pipewire_v4l2_c_args = [
   '-fvisibility=hidden',
 ]
 
-libv4l2_path = modules_install_dir / 'v4l2'
-libv4l2_path_dlopen = modules_install_dir_dlopen / 'v4l2'
+libv4l2_path = get_option('libv4l2-path')
+if libv4l2_path == ''
+  libv4l2_path = modules_install_dir / 'v4l2'
+  libv4l2_path_dlopen = modules_install_dir_dlopen / 'v4l2'
+else
+  libv4l2_path_dlopen = libv4l2_path
+endif
 
 tools_config = configuration_data()
 tools_config.set('LIBV4L2_PATH', libv4l2_path_dlopen)
-- 
GitLab

